log4j-core/src/main/java/org/apache/logging/log4j/core/impl/ThrowableProxyRenderer.java

```diff
@@ -148,11 +154,14 @@ private static void formatEntry(final ExtendedStackTraceElement extStackTraceEle
         textRenderer.render(lineSeparator, sb, "Text");
     }
 
-    private static boolean ignoreElement(final StackTraceElement element, final List<String> ignorePackages) {
-        if (ignorePackages != null) {
-            final String className = element.getClassName();
-            for (final String pkg : ignorePackages) {
-                if (className.startsWith(pkg)) {
+    private static boolean elementStartsWith(final StackTraceElement element, final List<String> packagesOrStartFrames, boolean includeMethodName) {
```

Comment:
Sorry, I am allergic to these kind of unions. Why not simply having two arguments: `packages` and `startFrames`? (Please mind the `final` modifiers.)